### PR TITLE
Chore/logging

### LINKED ISSUE #152: log user id in access log
would be great to be able to decode jwt token and get user id and inject to access log

### PR DESCRIPTION
**Pull request by frickjack**
* log client IP address (not ELB), and forward to proxied services
* log user (from JWT token)

resolve #152

### CODE REVIEW

File: kube/services/revproxy/00nginx-config.yaml [hunk 1]
```diff
@@ -8,12 +8,72 @@ metadata:
...
+        var user = "unknown@unknown";
+
+        if (token) {
+          user = token;
+          var raw = atob((token.split('.')[1] || "").replace('-', '+').replace('_', '/'));
+          if (raw) {
+            try {
+              var data = JSON.parse(raw);
+              if (data && data.context && data.context.user && data.context.user.name) {
+                user = data.context.user.name;
```
Comment (philloooo) [inline kube/services/revproxy/00nginx-config.yaml]: can we log both username and user id? (`data.sub`)

File: kube/services/revproxy/00nginx-config.yaml [hunk 2]
```diff
@@ -80,6 +150,8 @@ data:
           #   not set headers ... http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_set_header
           #
           proxy_set_header   Authorization "$access_token";
+          proxy_set_header   X-Forwarded-For "$http_x_forwarded_for";
+          proxy_set_header   X-UserId "$userid";
           #add_header X-debug-auth "$access_token"; #- only do this for quick and dirty client-side debugging
```
Comment (philloooo) [inline kube/services/revproxy/00nginx-config.yaml]: can remove this now?

File: kube/services/revproxy/00nginx-config.yaml [hunk 3]
```diff
@@ -102,47 +174,43 @@ data:
           }
           
           location / {
-              set $portal_service http://portal-service.default.svc.cluster.local;
-              if ($cookie_portal_service ~ ^[\w-]+$) {
```
Comment (philloooo) [inline kube/services/revproxy/00nginx-config.yaml]: why is this removed?

Comment (frickjack) [reply kube/services/revproxy/00nginx-config.yaml]: I want to replace with changes so we can deploy the whole commons into a namespace rather than just do the portal-switch with a cookie thing - which I don't think has turned out to be too useful ...?

File: kube/services/revproxy/00nginx-config.yaml (hunk 2 above)
Comment (frickjack) [reply kube/services/revproxy/00nginx-config.yaml]: Need to pass the IP address data and user data on to the child services, so they can log it ...
